doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -2,7 +2,615 @@
+- [Sample client application](#sample-client-application)
+    - [Preparation](#preparation)
+    - [Application code](#application-code)
+    - [Retrieving collected data](#retrieving-collected-data)
+    - [Changing sample period](#changing-sample-period)
+- [Next steps](#next-steps)
+    - [Profiling and grouping](#profiling-and-grouping)
+    - [Using notifications](#using-notifications])
+    - [Events](#events)
+    - [Distibuting operation data](#distibuting-operation-data)
+- [Troubleshooting](#troubleshooting)
+- [Further reading](#further-reading)
+
+# Prerequisites
+
+To register a new application within a fresh Kaa server installation,
+you need to create users with the [tenant administrator]({{root_url}}/Administration-guide/Tenants-and-applications-management/#tenant-admin) and [tenant developer]({{root_url}}/Administration-guide/Tenants-and-applications-management/#tenant-developer) roles.
+The tenant administrator is responsible for creating new applications in Kaa,
+and the tenant developer configures and generates SDKs for those applications.
+We suggest that you use Kaa Sandbox, which has a tenant administrator and tenant
+developer users already created.
+
+# Application description
+
+Application created trough this guide covers the basic case of collecting
+temperature data from a sensor. Usually, simple sensors stream data at constant
+rate, no matter what. However, rising demand of mobility and power efficiency dictates own
+rules of game. It is not viable to keep sensors streaming all the time, especially
+if a data is not required at this moment. That's a point where we can save power by
+configuring sample period.
+
+To satisfy application requirements two of main Kaa features are used:
+
+- **Data Collection** feature that allows to send data from endpoints
+to the Kaa server. In this example Data Collection is used to reliably transmit
+temperature samples from sensor to the Kaa.
+
+- **Configuration** feature that allows to broadcast configuration parameters
+from the Kaa server to Kaa endpoints. In this guide Configuration is used
+to pass a sample period of the temperature to endpoint.
+
+# Adding application
+
+To add an application, proceed as follows:
+
+1. Open the [Kaa admin UI in your browser](http://127.0.0.1:8080) and log in
+as a tenant administrator (user/password: admin/admin123).
+1. Select **Applications** on the navigation panel on the left side and, in the
+Applications window that opens, click **Add application**.
+1. In the Add application window, enter the application name and then click **Add**.
+For this guide we will use **Trustful** credential service and our application will be
+named "My First Kaa Application"
+
+    ![Admin Console](attach/new_app.png)
+
+After the application has been added, you may log out. We will not be using
+the tenant administrator role in this guide anymore.
+
+# Creating schemas
+
+The application that you have created in the previous step already includes
```

Comment:
Link to the previous step is needed for those who got here in an unexpected way.